include/swift/Basic/Compiler.h

```diff
@@ -103,4 +103,12 @@
 #define SWIFT_CRASH_BUG_REPORT_MESSAGE \
   "Please " SWIFT_BUG_REPORT_MESSAGE_BASE " and the crash backtrace."
 
+#ifdef NDEBUG
+#define SWIFT_ASSERT_ONLY_DECL(X)
```

Comment:
Can you add a comment providing a description in words and an example of how this is supposed to be used?

Reply:
```
// Conditionally exclude declarations or statements that are only needed for
// assertions from release builds (NDEBUG) without cluttering the surrounding
// code by #ifdefs.
//
// struct DoThings  {
//   SWIFT_ASSERT_ONLY_DECL(unsigned verifyCount = 0);
//   DoThings() {
//     SWIFT_ASSERT_ONLY(verifyCount = getNumberOfThingsToDo());
//   }
//   void doThings() {
//     do {
//       // ... do each thing
//       SWIFT_ASSERT_ONLY(--verifyCount);
//     } while (!done());
//     assert(verifyCount == 0 && "did not do everything");
//   }
// };
```

Reply:
Thanks.